File: frontend/src/api/news.ts
import axios from 'axios'

const api = axios.create({
  baseURL: '/api',
  timeout: 30000,
})

export interface NewsArticle {
  id: number
  title: string
  content?: string
  summary?: string
  pub_time?: string
  source?: string
  url: string
  created_at: string
  importance_score?: number
  is_incremental?: boolean
  expectation_diff?: string
  key_points?: string[]
  category?: string
  final_score?: number
  news_type?: string
  is_processed: boolean
  processed_at?: string
  notes?: string
}

export interface NewsListResponse {
  total: number
  page: number
  page_size: number
  items: NewsArticle[]
}

export interface Stats {
  total: number
  urgent_count: number
  important_count: number
  general_count: number
  noise_count: number
  unprocessed_count: number
  incremental_count: number
  avg_score?: number
}

export interface ScrapeRequest {
  days: number
}

export const getNewsList = async (params: {
  page?: number
  page_size?: number
  news_type?: string
  time_range?: string
  sort_by?: string
  is_incremental?: boolean
}): Promise<NewsListResponse> => {
  const response = await api.get('/news', { params })
  return response.data
}

export const getNewsDetail = async (id: number): Promise<NewsArticle> => {
  const response = await api.get(`/news/${id}`)
  return response.data
}

export const getStats = async (): Promise<Stats> => {
  const response = await api.get('/news/stats/summary')
  return response.data
}

export const scrapeNews = async (data: ScrapeRequest): Promise<void> => {
  await api.post('/news/scrape', data)
}

export const analyzeNews = async (id: number): Promise<void> => {
  await api.post(`/news/analyze/${id}`)
}

export const analyzeAllNews = async (params: { unprocessed_only?: boolean }): Promise<void> => {
  await api.post('/news/analyze-all', null, { params })
}

export const updateNewsType = async (
  id: number,
  data: { news_type?: string; is_processed?: boolean; notes?: string }
): Promise<void> => {
  await api.put(`/news/${id}/type`, data)
}

export const deleteNews = async (id: number): Promise<void> => {
  await api.delete(`/news/${id}`)
}

File: backend/app/schemas/news.py
from datetime import datetime
from typing import Optional, List, Any
from pydantic import BaseModel, Field
from enum import Enum


class NewsTypeEnum(str, Enum):
    URGENT = "紧急"
    IMPORTANT = "重点关注"
    GENERAL = "一般参考"
    NOISE = "已处理噪音"


class ExpectationDiffEnum(str, Enum):
    EXCEED = "超预期"
    MEET = "符合预期"
    BELOW = "低于预期"
    UNCLEAR = "无法判断"


class PageParams(BaseModel):
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
    news_type: Optional[NewsTypeEnum] = None
    time_range: str = Field(default="1d", pattern="^(1d|3d|7d|30d)$")
    sort_by: str = Field(default="final_score", pattern="^(final_score|pub_time|importance_score)$")
    is_incremental: Optional[bool] = None


class NewsArticleCreate(BaseModel):
    title: str
    content: Optional[str] = None
    summary: Optional[str] = None
    pub_time: Optional[datetime] = None
    source: Optional[str] = None
    url: str


class NewsArticleUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    summary: Optional[str] = None
    is_processed: Optional[bool] = None
    notes: Optional[str] = None
    news_type: Optional[str] = None


class NewsAnalysisResult(BaseModel):
    importance_score: int = Field(ge=0, le=100)
    is_incremental: bool
    expectation_diff: ExpectationDiffEnum
    key_points: List[str] = []
    category: str


class NewsArticleResponse(BaseModel):
    id: int
    title: str
    content: Optional[str] = None
    summary: Optional[str] = None
    pub_time: Optional[datetime] = None
    source: Optional[str] = None
    url: str
    created_at: datetime
    importance_score: Optional[int] = None
    is_incremental: Optional[bool] = None
    expectation_diff: Optional[str] = None
    key_points: Optional[List[str]] = None
    category: Optional[str] = None
    final_score: Optional[float] = None
    news_type: Optional[str] = None
    is_processed: bool = False
    processed_at: Optional[datetime] = None
    notes: Optional[str] = None

    class Config:
        from_attributes = True


class NewsArticleListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: List[NewsArticleResponse]


class ScrapeRequest(BaseModel):
    days: int = Field(default=3, ge=1, le=7)


class StatsResponse(BaseModel):
    total: int
    urgent_count: int
    important_count: int
    general_count: int
    noise_count: int
    unprocessed_count: int
    incremental_count: int
    avg_score: Optional[float] = None
